lib/node_modules/@stdlib/string/len/lib/is-surrogate.js

```diff
@@ -0,0 +1,55 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2020 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MAIN //
+
+/**
+* Returns whether the string has a surrogate pair at a position.
+*
+* @param {string} str - input string
+* @param {number} pos - position in string
+* @returns {boolean} whether the string has a surrogate pair at a position
+*
+* @example
+* var out = isSurrogate( 'last man standing', 4 );
+* // returns false
+*
+* @example
+* var out = isSurrogate( 'presidential election', 8 );
+* // returns false
+*
+* @example
+* var out = isSurrogate( 'अनुच्छेद', 2 );
+* // returns false
+*
+* @example
+* var out = isSurrogate( '🌷', 0 );
+* // returns true
+*/
+function isSurrogate( str, pos ) {
+	return str.charCodeAt( pos ) >= 0xd800 && str.charCodeAt( pos ) <= 0xdbff &&
```

Comment:
Can probably cache the char codes, rather than invoke `charCodeAt` twice for each position. For example,

```javascript
    var ch1 = str.charCodeAt( pos );
    var ch2 = str.charCodeAt( pos+1 );
    return ...;
```